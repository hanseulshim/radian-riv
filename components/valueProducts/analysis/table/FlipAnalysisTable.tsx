import React, { useState, useCallback } from 'react'
import Table from 'components/common/Table'
import { formatPercent, formatPrice } from 'utils'

interface Props {
  tableData: any[]
  flipType: string
}

export default function FlipAnalysisTable({ tableData, flipType }: Props) {
  const [data, setData] = useState([])
  const [columns, setColumns] = useState([])

  const fetchData = useCallback(async () => {
    const width = 90
    const parseValue = (key: string, value: number) => {
      return key.includes('Number')
        ? value
        : key.includes('%') || key.includes('Cap')
        ? formatPercent(value)
        : formatPrice(value)
    }
    setColumns([
      {
        Header: flipType,
        columns: [
          {
            Header: 'Market Area Flip Report',
            accessor: 'report',
            align: 'right'
          },
          {
            Header: '0-90',
            accessor: row => parseValue(row.report, row['0-90']),
            width,
            align: 'right'
          },
          {
            Header: '91-180',
            accessor: row => parseValue(row.report, row['91-180']),
            width,
            align: 'right'
          },
          {
            Header: '180-270',
            accessor: row => parseValue(row.report, row['180-270']),
            width,
            align: 'right'
          },
          {
            Header: '271-365',
            accessor: row => parseValue(row.report, row['271-365']),
            width,
            align: 'right'
          }
        ]
      }
    ])
    setData(tableData)
  }, [tableData])

  return (
    <div style={{ marginRight: flipType.includes('SOLD') ? '10px' : 0 }}>
      <Table columns={columns} data={data} fetchData={fetchData} width={520} />
    </div>
  )
}
