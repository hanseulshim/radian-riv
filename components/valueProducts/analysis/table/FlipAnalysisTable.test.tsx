import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import FlipAnalysisTable from './FlipAnalysisTable'

describe('FlipAnalysisTable', () => {
  let component
  beforeAll(() => {
    const props = {
      tableData: [],
      flipType: 'test'
    }
    component = shallow(<FlipAnalysisTable {...props} />)
  })
  it('get table container', () => {
    expect(component.find('div')).toHaveLength(1)
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
