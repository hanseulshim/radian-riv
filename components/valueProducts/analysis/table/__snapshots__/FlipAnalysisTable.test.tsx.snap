// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlipAnalysisTable should match snapshot 1`] = `
<div
  style={
    Object {
      "marginRight": 0,
    }
  }
>
  <Table
    columns={Array []}
    data={Array []}
    fetchData={[Function]}
    width={520}
  />
</div>
`;
